Sources/TuistSigning/SigningCipher.swift

```diff
@@ -0,0 +1,134 @@
+        else { throw SigningCipherError.failedToEncrypt }
+        return data
+    }
+
+    /// Decrypts `data`
+    /// - Parameters:
+    ///     - data: Data to decrypt
+    ///     - masterKey: Master key data
+    /// - Returns: Decrypted data
+    private func decryptData(_ data: Data, masterKey: Data) throws -> Data {
+        guard
+            let encodedString = String(data: data, encoding: .utf8),
+            let dividerIndex = encodedString.firstIndex(of: "-"),
+            let iv = Data(base64Encoded: String(encodedString.prefix(upTo: dividerIndex)))
+        else { throw SigningCipherError.failedToEncrypt }
+
+        let dataToDecrypt = Data(base64Encoded: String(encodedString.suffix(from: dividerIndex).dropFirst()))
+        let aesCipher = try AES(key: masterKey.bytes, blockMode: CTR(iv: iv.bytes), padding: .noPadding)
+        guard let decryptedData = try dataToDecrypt?.decrypt(cipher: aesCipher) else { throw SigningCipherError.failedToEncrypt }
+        return decryptedData
+    }
+
+    /// - Returns: Files we want encrypt/decrypt along with master key data
+    private func signingData(at path: AbsolutePath) throws -> (signingKeyFiles: [AbsolutePath], masterKey: Data) {
+        guard
+            let rootDirectory = rootDirectoryLocator.locate(from: path)
+        else { throw SigningCipherError.rootDirectoryNotFound(path) }
+        let signingDirectory = rootDirectory.appending(components: Constants.tuistDirectoryName, Constants.signingDirectoryName)
+        let masterKey = try self.masterKey(from: signingDirectory)
+        // Find all files in `signingDirectory` with the exception of "master.key"
+        let signingKeyFiles = FileHandler.shared.glob(signingDirectory, glob: "**/*")
+            .filter { $0.pathString != signingDirectory.appending(component: "master.key").pathString }
+            .filter { !FileHandler.shared.isFolder($0) }
+        return (signingKeyFiles: signingKeyFiles, masterKey: masterKey)
+    }
+
+    /// - Returns: Master key data
+    private func masterKey(from signingDirectory: AbsolutePath) throws -> Data {
+        let masterKeyFile = signingDirectory.appending(component: "master.key")
+        guard FileHandler.shared.exists(masterKeyFile) else { throw SigningCipherError.masterKeyNotFound(masterKeyFile) }
+        let plainMasterKey = try FileHandler.shared.readFile(masterKeyFile)
+        return plainMasterKey.sha256()
+    }
+
+    /// - Returns: Data of generated initialization vector
+    private func generateIv() throws -> Data {
+        let blockSize = 16
+        var iv = Data(repeating: 0, count: blockSize)
+        let result = iv.withUnsafeMutableBytes { bytes -> Int32 in
+            guard let baseAddress = bytes.baseAddress else { return 0 }
+            return SecRandomCopyBytes(kSecRandomDefault, blockSize, baseAddress)
+        }
+        guard result == errSecSuccess else { throw SigningCipherError.failedToEncrypt }
```

Comment:
This is where you could pass the result, so we can print it out to the end-user. The error code might be good enough, but [KeychainAccess](https://github.com/kishikawakatsumi/KeychainAccess/blob/master/Lib/KeychainAccess/Keychain.swift#L1768-L2994) went one step further and provided a description for each message - I will leave it up to you to consider.

Reply:
I am [trying](https://github.com/tuist/tuist/pull/1127/files#diff-eb7af229bd4931d822a9924dd4e81f9aR139) to parse the error and if that fails, I print out the error code